fw/hpack.c

```diff
@@ -3488,6 +3488,59 @@ tfw_hpack_hdr_expand(TfwHttpResp *__restrict resp, TfwStr *__restrict hdr,
 	if (!hdr)
 		return -EINVAL;
 
+	// compact str with sparse chunks in place
+	// e.g. changes
+	// "f" "o" ":" " " "b" "a"
+	// into
+	// "fo" ": " "ba"
+	// it's different from `tfw_http_hdr_split()`, which only splits
+	// name chunks and value chunks into two strs, no compact.
+	if (hdr->flags & TFW_STR_TRAILER) {
```

Comment:
Why we should do it? Is this transformation necessary according RFC? If yes please add comment. And we don't need this part of code. I think that we can do it during parsing trailer headers in http_parser.c. @const-t how do you think?

Reply:
This conversion only makes sense for our implementation to satisfy the needs of the header encoding below.

Reply:
We don't need this part of code if we use pool.

Reply:
[ditto](https://github.com/tempesta-tech/tempesta/pull/2173/commits/bb1589a70478b69aff2a7a455c72da865a0ec4bd).